sc/social/like/browser/canonicalurl.py

```diff
@@ -106,9 +107,7 @@ def update_canonical_url(self, data):
         logger.info(u'{0} objects will have their canonical URL updated'.format(total))
 
         for obj in get_valid_objects(results):
-            # FIXME: we're currently ignoring the Plone site id
-            #        https://github.com/collective/sc.social.like/issues/119
-            path = '/'.join(obj.getPhysicalPath()[2:])
+            path = get_path_to_virtual_path(self.request, obj)
```

Comment:
we need to fix this on a different way: this form is only used to assign a canonical URL to objects created before the package was installed/upgraded.

now imagine the following case: Plone site with the id visible to the end user migrating to a site where the id is no longer available, that is from `http://www.example.org/Plone` to `http://www.example.org/`.

we're currently not covering this case and the only solution I can think about is to add a `Bool` field to enforce the use of the site id or a `ASCIILine` field that lets you add an arbitrary string.

we need to think about this.